### DISCUSSION TITLE
Copilot should not block execution of code in VSCode Jupyter for dozens of seconds while doing nothing useful

### DISCUSSION THREAD

**Discussion by JasonGross** [community/community#121204] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Consider this egregiously slow behavior (and I've sometimes seen things be 10x as slow):
![2024-04-25-slow-ctrl-enter-vscode](https://github.com/microsoft/vscode-jupyter/assets/396076/2036bdc4-8da7-4982-822d-de53feb7dd6a)

The performance issues [here](https://github.com/microsoft/vscode-jupyter/issues/14459#issuecomment-2078873976) seem like they might only show up when copilot is enabled.

I say that Copilot is doing nothing useful because there is effectively no CPU usage, as I note [here](https://github.com/microsoft/vscode-jupyter/issues/14459#issuecomment-2079140990).

Related: https://github.com/orgs/community/discussions/10053

**Comment by jiagengliu** [comment]
This happens to me in 1.89 as well, even with Copilot disabled. After starting a new VSCode session, it tends to be fine, but after a while, like 20 minutes, it suddenly takes a long time to start executing the code, similar to your situation.

**Reply by ed1d1a8d** [reply]
Am running into the same issue! My setup is using Remote - SSH with a linux remote, and after 20 minutes, I also run into cells starting to hang randomly. I do notice vscode having a spike in CPU usage when this happens though.
